spec-main/api-web-contents-spec.ts

```diff
@@ -1191,6 +1195,27 @@ describe('webContents module', () => {
       expect(currentRenderViewDeletedEmitted).to.be.false('current-render-view-deleted was emitted');
     });
 
+    it('does not emit current-render-view-deleted when speculative RVHs are deleted and nativeWindowOpen is set to true', async () => {
+      const parentWindow = new BrowserWindow({ show: false, webPreferences: { nativeWindowOpen: true } });
+      let currentRenderViewDeletedEmitted = false;
+      let childWindow:BrowserWindow;
+      const destroyed = emittedOnce(parentWindow.webContents, 'destroyed');
+      const renderViewDeletedHandler = () => {
+        currentRenderViewDeletedEmitted = true;
+      };
+      app.once('browser-window-created', (event, window) => {
+        childWindow = window;
+        window.webContents.on('current-render-view-deleted' as any, renderViewDeletedHandler);
+      });
+      parentWindow.loadURL(`${serverUrl}/first-window-open`);
+      setTimeout(() => {
```

Comment:
Can we listen on navigation events on the childWindow before closing the parentWindow ? In this case it would be `did-fail-load` since the second url is a invalid url. I am bit worried about `setTimeout` being a potential for flaky tests.

Reply:
I've checked that `did-fail-load` is not emitted for this url. I've tried to find another event which we could use but with no luck so far.

Reply:
@deepak1556 FWIW... it appears you are right as WOA is now failing on this test fairly regularly:
https://github.visualstudio.com/electron/_build/results?buildId=85266&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=5caf77c8-9b10-50ef-b5c7-ca89c63e1c86&l=2139